### PR TITLE
feat(project): add `include` field

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

As part of the RFC https://github.com/biomejs/biome/discussions/180, this PR adds a new configuration called `include`.

This new field is added to the following:
- `files.include`
- `linter.include`
- `formatter.include`
- `organizeImports.include`

The way it works is that Biome will handle only the files that match those patterns. When `include` and `ignore` are both defined for the same tool, the field `ignore` takes precedence.

## Test Plan

Added various test cases for the majority of the commands.

### CODE REVIEW

**Comment by Conaclos** [comment]
> When include and ignore are both defined for the same tool, the field ignore takes precedence.

Just for clarifying. Do you mean that if a file is both matched by `include` and `ignore`, then it is ignored? The other files that are not matched by `include` and `ignore` are considered ignored?

**Comment by ematipico** [comment]
> > When include and ignore are both defined for the same tool, the field ignore takes precedence.
> 
> Just for clarifying. Do you mean that if a file is both matched by `include` and `ignore`, then it is ignored? 

Yes, `ignore` has precedence over `include` and that's the expected behaviour by this feature.

> The other files that are not matched by `include` and `ignore` are considered ignored?

Yes, that's the idea. This means that if you use `"include": []`, no files are processed.

> 
> By the way, should we plan to rename `ignore` to `exclude`?

I thought about it, but it's not part of this PR.

**Comment by ematipico** [comment]
Website preview https://biome-website-hrddoozmo-ematipico.vercel.app/reference/configuration/#filesinclude

File: website/src/content/docs/reference/configuration.mdx [hunk 1]
```diff
@@ -165,6 +165,42 @@ match these patterns.
...
+```json
+{
+  "files": {
+    "include": ["scripts/**/*.js", "src/**/*.js"],
+    "exclude": ["scripts/**/*.js"]
+  }
+}
+```
+
+Only the files that match the patter `src/**/*.js` will be handled, while the files that match the pattern
```
Comment (Conaclos) [inline website/src/content/docs/reference/configuration.mdx:201]:
```suggestion
Only the files that match the pattern `src/**/*.js` will be handled, while the files that match the pattern
```

**Review by Conaclos — APPROVED** [review]

**Comment by victor-teles** [comment]
@ematipico I'm having a weird behavior

When i include + ignore same file, all my workspace files still included,
But i only want to include `ecosystem.config.js` and `vitest.config.ts`

In that case, only `vitest.config.ts` should be included, right?

image

**Comment by ematipico** [comment]
I'll look into it, thank you for testing it

**Comment by ematipico** [comment]
@victor-teles the latest commit should fix your case, I added a test case. Thank you for your feedback!

**Pull request merged**